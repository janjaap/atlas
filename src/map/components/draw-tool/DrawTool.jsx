import React from 'react';
import PropTypes from 'prop-types';

import './_draw-tool.scss';
import drawToolConfig from '../../services/draw-tool/draw-tool-config';
import ToggleDrawing from '../toggle-drawing/ToggleDrawing';
import ShapeSummary from '../shape-summary/ShapeSummary';
import PointsAvailable from '../points-available/PointsAvailable';

const DrawTool = ({
  drawingMode,
  isEnabled,
  onClearDrawing, // TODO: rename
  shapeDistanceTxt,
  shapeMarkers,
  toggleDrawing
}) => {
  const markersLeft = drawToolConfig.MAX_MARKERS - shapeMarkers;
  return (
    <section className="draw-tool">
      <ToggleDrawing
        drawingMode={drawingMode}
        shapeMarkers={shapeMarkers}
        toggleDrawing={toggleDrawing}
      />
      {
        !isEnabled && shapeMarkers === 2
        &&
        <ShapeSummary
          shapeDistanceTxt={shapeDistanceTxt}
          onClearDrawing={onClearDrawing}
        />
      }
      {
        markersLeft <= drawToolConfig.MARKERS_LEFT_WARNING &&
        isEnabled &&
        <PointsAvailable
          markersLeft={markersLeft}
          drawingMode={drawingMode}
        />
      }
    </section>
  );
};

DrawTool.propTypes = {
  drawingMode: PropTypes.string.isRequired,
  isEnabled: PropTypes.bool.isRequired,
  onClearDrawing: PropTypes.func.isRequired,
  shapeDistanceTxt: PropTypes.string.isRequired,
  shapeMarkers: PropTypes.number.isRequired,
  toggleDrawing: PropTypes.func.isRequired
};

export default DrawTool;
