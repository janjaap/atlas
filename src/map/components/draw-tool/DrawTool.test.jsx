import React from 'react';
import { shallow } from 'enzyme';

import DrawTool from './DrawTool';
import drawToolConfig from '../../services/draw-tool/draw-tool-config';

describe('DrawTool', () => {
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      isEnabled: false,
      toggleDrawing: jest.fn(),
      onClearDrawing: jest.fn(),
      drawingMode: drawToolConfig.DRAWING_MODE.NONE,
      shapeMarkers: 0,
      shapeDistanceTxt: 'myDistanceTxt'
    };
  });

  it('should render toggle drawing', () => {
    const wrapper = shallow(<DrawTool {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render summary for 2 points', () => {
    const wrapper = shallow((
      <DrawTool
        {...defaultProps}
        shapeMarkers={2}
      />
    ));
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render summary when tool is enabled', () => {
    const wrapper = shallow((
      <DrawTool
        {...defaultProps}
        isEnabled
        shapeMarkers={2}
      />
    ));
    expect(wrapper).toMatchSnapshot();
  });

  it('should render points available if near max points', () => {
    const wrapper = shallow((
      <DrawTool
        {...defaultProps}
        isEnabled
        shapeMarkers={drawToolConfig.MAX_MARKERS - 2}
      />
    ));
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render points available if tool is disabled', () => {
    const wrapper = shallow((
      <DrawTool
        {...defaultProps}
        shapeMarkers={10}
      />
    ));
    expect(wrapper).toMatchSnapshot();
  });
});
