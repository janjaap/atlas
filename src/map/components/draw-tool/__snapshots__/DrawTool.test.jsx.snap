// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawTool should not render points available if tool is disabled 1`] = `
<section
  className="draw-tool"
>
  <ToggleDrawing
    drawingMode="none"
    shapeMarkers={10}
    toggleDrawing={[MockFunction]}
  />
</section>
`;

exports[`DrawTool should not render summary when tool is enabled 1`] = `
<section
  className="draw-tool"
>
  <ToggleDrawing
    drawingMode="none"
    shapeMarkers={2}
    toggleDrawing={[MockFunction]}
  />
</section>
`;

exports[`DrawTool should render points available if near max points 1`] = `
<section
  className="draw-tool"
>
  <ToggleDrawing
    drawingMode="none"
    shapeMarkers={38}
    toggleDrawing={[MockFunction]}
  />
  <PointsAvailable
    drawingMode="none"
    markersLeft={2}
  />
</section>
`;

exports[`DrawTool should render summary for 2 points 1`] = `
<section
  className="draw-tool"
>
  <ToggleDrawing
    drawingMode="none"
    shapeMarkers={2}
    toggleDrawing={[MockFunction]}
  />
  <ShapeSummary
    onClearDrawing={[MockFunction]}
    shapeDistanceTxt="myDistanceTxt"
  />
</section>
`;

exports[`DrawTool should render toggle drawing 1`] = `
<section
  className="draw-tool"
>
  <ToggleDrawing
    drawingMode="none"
    shapeMarkers={0}
    toggleDrawing={[MockFunction]}
  />
</section>
`;
